app.go

```diff
@@ -260,20 +260,22 @@ func (t stopTimeoutOption) String() string {
 	return fmt.Sprintf("fx.StopTimeout(%v)", time.Duration(t))
 }
 
-// withLogger will stay private until we export fxlog.Logger once
-// we finalize the API.
-func withLogger(l fxlog.Logger) Option {
-	return withLoggerOption{l}
+// WithLogger redirects the logging output to the instance of fx.Logger
+// provided by the constructor.
+func WithLogger(constructor interface{}) Option {
```

Comment:
I see, is there harm in leaving this public so we can keep on using this in tests and elsewhere and isolate exporting fxlog.Logger?

Reply:
The reason I'd want to unexport is because it's currently unusable. As a user of Fx, there's no valid way for me to call this function right now.

That said, if you think we won't release this change until we also finish exporting fxlog.Logger, I'm okay with keeping it exported in this PR.

Reply:
Yea, I think there isn't a point in releasing until this API is usable, so exporting fxlog.Logger should be a blocker for next release.